app/extensions/tus/__init__.py

```diff
@@ -65,12 +67,27 @@ def _tus_file_handler(upload_file_path, filename, req, app):
 
     if not os.path.exists(dir):
         os.makedirs(dir)
-    log.info('Tus finished uploading: %r in dir %r.' % (filename, dir))
-    os.rename(upload_file_path, os.path.join(dir, filename))
+    log.debug('Tus finished uploading: %r in dir %r.' % (filename, dir))
+    filepath = os.path.join(dir, filename)
+    os.rename(upload_file_path, filepath)
+
+    # Store the original filename as metadata next to the file
+    metadata_filepath = tus_get_metadata_filepath(filepath)
+    with open(metadata_filepath, 'w') as metadata_file:
+        metadata = {
+            'filename': escape(secure_filename(original_filename)),
```

Comment:
since it is run thru `secure_filename()`, arent you _not_ storing the **original filename** here?  is there any reason you are sanitizing something that is just being stored as json?

Reply:
I don't want to accidentally store insensitive metadata for things like XSS.  If we trust that the serialization code is going to sanitize user inputs for web-based attacks, we can remove this condition.  This is purely for showing the filename to the user, so we can have complete control over sanitizing now that the actual on-disk storage is disassociated with the security threat.

Reply:
Ok, I'm reverting this sanitizing.